app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivity.java

```diff
@@ -273,13 +290,49 @@ public void onClick(View v) {
     }
 
     public void goToIndividualChat(View view) {
-        Intent intent = new Intent(ActivityDescriptionActivity.this, ChatActivity.class);
-        String activityChatId = act.getActivityId() + " - chatId";
-        intent.putExtra("ACTIVITY_CHAT_ID", activityChatId);
-        String activityTitle = act.getTitle();
-        intent.putExtra("ACTIVITY_TITLE", activityTitle);
-        startActivity(intent);
+        // Allowing the access to the chat only if the user is registered to the activity
+        if (act.getParticipantId().contains(userId)) {
+            Intent intent = new Intent(ActivityDescriptionActivity.this, ChatActivity.class);
+            String activityChatId = act.getActivityId() + " - chatId";
+            intent.putExtra("ACTIVITY_CHAT_ID", activityChatId);
+            String activityTitle = act.getTitle();
+            intent.putExtra("ACTIVITY_TITLE", activityTitle);
+            startActivity(intent);
+        } else {
+            Toast.makeText(ActivityDescriptionActivity.this, "Please register if you want to access the chat!", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private void loadActivityHeaderPicture() {
+        activityImage = findViewById(R.id.activity_image_description);
+        progressBar = findViewById(R.id.progress_bar_activity_description);
+        progressBar.setVisibility(View.VISIBLE);
+        storageReference = FirebaseStorage.getInstance().getReference();
+        if (act != null) {
+            imagePath = "activities/"+act.getActivityId()+"/activityImage.jpg";
+            StorageReference imageRef = storageReference.child(imagePath);
```

Comment:
You don't store the image in the activity ?

Reply:
Right for the moment the image is solely stored on Firebase Storage. It is anyway linked (referenced with a unique path containing the sport activity ID) to the sport activity as its "header picture", but for the moment we don't have a local cache caching the "header pictures" of the sport activities! This is the next step.